presto-spi/src/test/java/io/prestosql/spi/block/BaseBlockEncodingTest.java

```diff
@@ -15,19 +15,87 @@
 
 import io.airlift.slice.DynamicSliceOutput;
 import io.prestosql.spi.type.Type;
+import org.testng.annotations.DataProvider;
 import org.testng.annotations.Test;
 
+import java.util.Arrays;
+import java.util.Random;
+import java.util.stream.IntStream;
+
 import static io.prestosql.spi.block.BlockTestUtils.assertBlockEquals;
 
 @Test(singleThreaded = true)
 public abstract class BaseBlockEncodingTest<T>
 {
+    private static final int[] RANDOM_BLOCK_SIZES = {2, 4, 8, 9, 16, 17, 32, 33, 64, 65, 1000, 1000000};
+
+    private final Random random = new Random(32167);
     private final BlockEncodingSerde blockEncodingSerde = new TestingBlockEncodingSerde();
 
     protected abstract Type getType();
 
     protected abstract void write(BlockBuilder blockBuilder, T value);
 
+    protected abstract T randomValue(Random random);
+
+    @Test
+    public void testEmpty()
+    {
+        roundTrip();
+    }
+
+    @Test
+    public void testSingleNull()
+    {
+        roundTrip((T) null);
```

Comment:
is `(T)` redundant here?

Reply:
This method has varargs. Only `null` here will be ambiguous  - don't know whether it's an array with a single `null` value or an actual `null`.